apps/remote_control/lib/lexical/remote_control/code_action/handlers/replace_remote_function.ex

```diff
@@ -0,0 +1,136 @@
+defmodule Lexical.RemoteControl.CodeAction.Handlers.ReplaceRemoteFunction do
+  alias Lexical.Ast
+  alias Lexical.Document
+  alias Lexical.Document.Changes
+  alias Lexical.Document.Edit
+  alias Lexical.Document.Range
+  alias Lexical.RemoteControl.CodeAction
+  alias Lexical.RemoteControl.CodeAction.Diagnostic
+  alias Sourceror.Zipper
+
+  @behaviour CodeAction.Handler
+
+  @impl CodeAction.Handler
+  def actions(%Document{} = doc, %Range{}, diagnostics) do
+    Enum.flat_map(diagnostics, fn %Diagnostic{} = diagnostic ->
+      with {:ok, module, function, arity, line_number} <- extract_function_and_line(diagnostic),
```

Comment:
Could you elaborate on this one?

Reply:
You're parsing the line number out of the message, but the diagnostic has a line number in its range field. 

https://github.com/lexical-lsp/lexical/blob/e1b9b86127c62db2a1f5919bc3b282472f33aa6b/apps/remote_control/lib/lexical/remote_control/code_action/diagnostic.ex#L8

I think that should indicate where the error occurs

Reply:
Actually, I'm already using diagnostic.range.start.line:

```elixir
defp extract_function_and_line(%Diagnostic{} = diagnostic) do
  with {:ok, module, function, arity} <- extract_function(diagnostic.message),
       {:ok, line_number} <- extract_line(diagnostic) do
    {:ok, module, function, arity, line_number}
  end
end

defp extract_line(%Diagnostic{} = diagnostic) do
  {:ok, diagnostic.range.start.line}
end
```

Reply:
yeah, but why bother doing that in a function that can fail? It's always there, you don't need to put it in one of the arms of a with. 
```elixir
with {:ok, module, function, arity} <- extract_function(diagnostic.message) do 
  {:ok, module, function, arity, diagnostic.range.start.line}
end
```
This is a small nitpick, it's not enough to hold up the PR.

Reply:
Oh true, it cannot fail. I think here we would need to use `case` since we have just one `with` binding. That made me consider simplifying it even more and putting it in the same function as the contents of `extract_function`.